src/Chunk.cpp

```diff
@@ -1885,7 +1885,13 @@ bool cChunk::AddClient(cClientHandle * a_Client)
 
 void cChunk::RemoveClient(cClientHandle * a_Client)
 {
-	m_LoadedByClient.erase(std::remove(m_LoadedByClient.begin(), m_LoadedByClient.end(), a_Client));
+	auto itr = std::remove(m_LoadedByClient.begin(), m_LoadedByClient.end(), a_Client);
+	// We should always remove at most one client.
+	ASSERT(std::distance(itr, m_LoadedByClient.end()) <= 1);
+	if (itr != m_LoadedByClient.end())
```

Comment:
Agreed, but there still isn't a reason for an explicit check. Assuming that the server *isn't* super-buggy (ah, what an assumption :P), it's not needed.